### PR TITLE
Add support for nRF52 DK platform

### PR DESCRIPTION
**Pull request by wbober**
This PR adds support for nRF52 SOC and nRF52 Development Kit from Nordic Semiconductor.

The nRF52 is a Bluetooth SoC which supports IPv6 over Bluetooth Low Energy. Therefore this port demonstrates how Contiki can be used for IPv6 over BLE communication.

The PR adds support for nRF52832 SoC and nRF52 DK. Examples and docummentation that demonstrate how to use the DK are also provided.

The following features have been implemented:
* Support for IPv6 over BLE using Contiki 6LoWPAN implementation
* Contiki system clock and rtimers (using 32kHz and 1MHz timers)
* UART driver
* Watchdog driver
* Hardware RNG
* Temperature sensor driver
* DK LED driver
* DK Buttons driver
* Segger Real Time Transfer (RTT) I/O support

This port is officially supported by Nordic Semiconductor. 

This PR depends on two eariler PR, which were already reviewed:
https://github.com/contiki-os/contiki/pull/1428
https://github.com/contiki-os/contiki/pull/1429

### CODE REVIEW

**Comment by wbober** [comment]
I've rebased this PR on the latest master.

**Comment by wbober** [comment]
What's is the status of this PR? Is there anything going on to get it merged?

**Comment by alignan** [comment]
I don't have the hardware to test, and have little bandwidth to review. I'm glad to review if I could get the DK.  Perhaps another merger has already worked with something similar?

**Comment by joakimeriksson** [comment]
We have some boards at SICS and I have discussed with the people using the platform to give feedback. Will push a bit more upcoming days so that we get some feedback.

**Comment by alignan** [comment]
At least code source seems properly formatted at a glance, if you can confirm you have uncrustified the contribution then it would be a less thing to look at :-)

**Comment by wbober** [comment]
@joakimeriksson Thanks for an update.
@alignan I've used the Eclipse formatter. I've read somewhere that the uncrustify tool doesn't handle all the cases correctly, right?

**Comment by alignan** [comment]
Just some corner cases like `while(...);`, easy to spot if you diff.

**Comment by johangas** [comment]
Hey, I've been working on the boards at SICS for a few weeks now as part of my master thesis.

So far, I've tested the timers, LEDs, buttons, RNG and the IPv6 over BLE. Overall, the port works as expected.

The BLE works fine but is missing the master role which is one of the main benefits of the platform. It would be great if it was implemented.

As I'm doing this type of review for the first time, I gladly accept tips on what should be investigated.

**Comment by wbober** [comment]
@johangas 

I've discussed the issue of master role and I was told that the IPSP supports it. It should not be difficult to add it to the port if you wish to make an attempt.

**Comment by alignan** [comment]
I'm normally concerned on three things:

* Is the `README` clear enough to get people started? is the documentation up to date and the installation of resources, toolchain, etc properly documented?
* Does the basic Contiki examples work? or there is a reason for which i.e `hello-world` won't work.
* Does the specific platform examples work? are there any tweaks required for the example to work?

Also there should be clearly stated who is the platform maintainer, in this case it is stated it will be Nordic, but a name/mail in the `README` file should be preferable instead (see the [jn516x](https://github.com/contiki-os/contiki/blob/master/platform/jn516x/README.md)).

**Comment by johangas** [comment]
@wbober 
I don't think there is time to do both what the thesis is about and the master role, unfortunately. I could look into it after I'm done but I will make no promises at this point.

@alignan 
> - Is the README clear enough to get people started? is the documentation up to date and the installation of resources, toolchain, etc properly documented?
It is clear. There is however a special case in the setup docs which i have informed @wbober about over email.
> - Does the basic Contiki examples work? or there is a reason for which i.e hello-world won't work.
Yeah, hello-world and timer work fine, for example.
> - Does the specific platform examples work? are there any tweaks required for the example to work?
Yes. A suggestion would be to make examples interfacing to Nordics smartphone apps.

**Comment by alignan** [comment]
Great, thanks!

@wbober are you going to be the official maintainer? could you please add this to the platform's documentation as well?

:+1:

**Comment by wbober** [comment]
@alignan Yes. This info is already in the readme.md

File: core/net/ipv6/sicslowpan.c [hunk 1]
```diff
@@ -169,6 +169,14 @@ void uip_log(char *msg);
 #define COMPRESSION_THRESHOLD 0
 #endif
 
+/** \brief Fixed size of a frame header. This value is
+ * used in case framer returns an error or if SICSLOWPAN_USE_FIXED_HDRLEN
+ * is defined.
+ */
+#ifndef SICSLOWPAN_FIXED_HDRLEN
+#define SICSLOWPAN_FIXED_HDRLEN 21
```
Comment (alignan) [inline core/net/ipv6/sicslowpan.c]: This is a minor change that really doesn't affect `core`, but normally one we would prefer in a different PR, but as it is already in a separate commit I don't object :+1:

**Pull request merged**